### DISCUSSION TITLE
Renovate offered Node 20.11.0 update before it was released

### DISCUSSION THREAD

**Discussion by broksonic21** [renovatebot/renovate#26579] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

Last evening (note: it's since been released so this no longer repros) we got PRs in all of our branches to bump this section of package.json to node v20.11.0:

```
  "volta": {
    "node": "20.10.0",
    "npm": "10.2.5"
  }
```

However, there was always an artifact problem, since 20.11.0 wasn't released yet (it was only tagged an hour ago: see [https://github.com/nodejs/node/issues/51423](https://github.com/nodejs/node/issues/51423) and [https://github.com/nodejs/node/releases/tag/v20.11.0](https://github.com/nodejs/node/releases/tag/v20.11.0)

This caused failures in both the renovatebot artifact update, but also all of our CIs.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: packageFiles with updates
{
  "baseBranch": "main"
  "config": {
    "npm": [
      {
        "deps": [
          {
            "commitMessageTopic": "Node.js",
            "currentValue": "20.10.0",
            "currentVersion": "20.10.0",
            "datasource": "github-tags",
            "depName": "node",
            "depType": "volta",
            "fixedVersion": "20.10.0",
            "isSingleVersion": true,
            "lockedVersion": null,
            "packageName": "nodejs/node",
            "prettyDepType": "volta",
            "registryUrl": "https://github.com",
            "sourceUrl": "https://github.com/nodejs/node",
            "versioning": "node",
            "warnings": [],
            "updates": [
              {
                "bucket": "latest",
                "newVersion": "v20.11.0",
                "newValue": "20.11.0",
                "releaseTimestamp": "2024-01-09T20:04:41.000Z",
                "newMajor": 20,
                "newMinor": 11,
                "updateType": "minor",
                "branchName": "renovatebot/node_npm"
              }
            ]
          },
          {
            "currentValue": "10.2.5",
            "currentVersion": "10.2.5",
            "datasource": "npm",
            "depName": "npm",
            "depType": "volta",
            "fixedVersion": "10.2.5",
            "homepage": "https://docs.npmjs.com/",
            "lockedVersion": null,
            "packageName": "npm",
            "prettyDepType": "volta",
            "registryUrl": "https://registry.npmjs.org",
            "sourceUrl": "https://github.com/npm/cli",
            "versioning": "npm",
            "warnings": [],
            "updates": []
          }
        ],
        "extractedConstraints": {
          "npm": ">=7"
        },
        "lockFiles": [
          "package-lock.json"
        ],
        "managerData": {
          "hasPackageManager": false,
          "npmLock": "package-lock.json",
          "packageJsonName": "aaron-test",
          "yarnZeroInstall": false
        },
        "npmrc": "***********",
        "packageFile": "package.json",
        "packageFileVersion": "0.6.1",
        "skipInstalls": true
      }
    ],
    "nvm": [
      {
        "deps": [
          {
            "currentValue": "20.10.0",
            "currentVersion": "20.10.0",
            "datasource": "github-tags",
            "depName": "node",
            "fixedVersion": "20.10.0",
            "isSingleVersion": true,
            "packageName": "nodejs/node",
            "registryUrl": "https://github.com",
            "sourceUrl": "https://github.com/nodejs/node",
            "versioning": "node",
            "warnings": [],
            "updates": [
              {
                "bucket": "latest",
                "newVersion": "v20.11.0",
                "newValue": "20.11.0",
                "releaseTimestamp": "2024-01-09T20:04:41.000Z",
                "newMajor": 20,
                "newMinor": 11,
                "updateType": "minor",
                "branchName": "renovatebot/node_npm"
              }
            ]
          }
        ],
        "packageFile": ".nvmrc"
      }
    ]
  }
}
```

```
DEBUG: branch status check result (branch="renovatebot/node_npm")
{
  "state": "failure"
  "statuses": [
    {
      "url": "https://api.github.com/repos/REDACTED/test/statuses/REDACTED",
      "avatar_url": "https://avatars.githubusercontent.com/in/REDACTED?v=4",
      "id": REDACTED,
      "node_id": "REDACTED",
      "state": "failure",
      "description": "Artifact file update failure",
      "target_url": null,
      "context": "renovate/artifacts",
      "created_at": "2024-01-10T00:44:47Z",
      "updated_at": "2024-01-10T00:44:47Z"
    }
  ]
}
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
How do you define "released"? Eg node website, docker hub?

**Reply by broksonic21** [reply]
@rarkins at minimum I'd go with:

* no artifact update problems after the PR comes out.

But more specific:
* Node release tagged as an actual release: it wasn't up at [https://github.com/nodejs/node/releases/tag/v20.11.0](https://github.com/nodejs/node/releases/tag/v20.11.0) at the time. and is still not up at [https://nodejs.org/en/about/previous-releases](https://nodejs.org/en/about/previous-releases) and [https://nodejs.org/en/download/](https://nodejs.org/en/download/)
* Able to install from nodejs.org - at the time, this failed with a 404: `curl -I https://nodejs.org/dist/v20.11.0/node-v20.11.0-darwin-arm64.tar.gz`, but that's now working

First one here is .nvmrc (for netlify usage), the second is volta settings, btw... neither worked at the time of PR creation, though

![image](https://github.com/renovatebot/renovate/assets/24306571/84a2d281-67b8-4143-9922-e058b7195916)

**Reply by rarkins** [reply]
Artifact failures can happen for many reasons. It's not a reliable judge of whether a release is valid or not.

However, we could add a feature which allows users to suppress PRs with update failures if any of the updates are less than X hours/days old.

Based on the logs for this repo where awe also have node engines, it seem we indeed use github-tags as the datasource:

```json
          {
            "commitMessageTopic": "Node.js",
            "currentValue": "^18.12.0 || >=20.0.0",
            "currentVersion": "v21.5.0",
            "datasource": "github-tags",
            "depName": "node",
            "depType": "engines",
            "packageName": "nodejs/node",
            "prettyDepType": "engine",
            "registryUrl": "https://github.com",
            "sourceUrl": "https://github.com/nodejs/node",
            "versioning": "node",
            "warnings": [],
            "updates": []
          }
```

However we need to compare releases to tags to make sure releases happen on 100% of versions which users will be interested in.

It could be better to index index https://nodejs.org/dist/

**Reply by viceice** [reply]
shouldn't it use the node-version datasource? 🤔

**Reply by viceice** [reply, marked as answer]
- #26918
